src/app/contexts/EventTrackingContext/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { createContext, useContext } from 'react';
+import { node } from 'prop-types';
+
+import { pageDataPropType } from '#models/propTypes/data';
+import { ServiceContext } from '#contexts/ServiceContext';
+import { RequestContext } from '#contexts/RequestContext';
+import { buildATIClickParams } from '#containers/ATIAnalytics/params';
+import useToggle from '#hooks/useToggle';
+import {
+  ARTICLE_PAGE,
+  FRONT_PAGE,
+  MEDIA_PAGE,
+  MOST_READ_PAGE,
+  MOST_WATCHED_PAGE,
+  INDEX_PAGE,
+  FEATURE_INDEX_PAGE,
+  MEDIA_ASSET_PAGE,
+  STORY_PAGE,
+  PHOTO_GALLERY_PAGE,
+  CORRESPONDENT_STORY_PAGE,
+} from '#app/routes/utils/pageTypes';
+
+export const EventTrackingContext = createContext({});
+
+export const EventTrackingContextProvider = ({ children, pageData }) => {
+  let pageIdentifier;
+  let platform;
+  let statsDestination;
+
+  const requestContext = useContext(RequestContext);
+  const serviceContext = useContext(ServiceContext);
+  const { enabled: eventTrackingIsEnabled } = useToggle('eventTracking');
+
+  if (pageData && eventTrackingIsEnabled) {
+    try {
```

Comment:
Is this try/catch just for logging?  This function seems to continue even if `buildATIClickParams` fails?

Maybe the exception handling / logging makes more sense inside `buildATIClickParams` rather than here?  Would simplify this function a bit

Reply:
Continuing on and not breaking the app is the behaviour we want in the event the pageData is not as expected.

Not sure if moving the exception handling to another function improves anything. Could do though

Reply:
I mean moving it into the `buildATIClickParams` function - the only thing in this `try` block here is a call to that function.  If that function is expected to throw exceptions, maybe that function should catch them, rather than dealing with it in this (already very complex) function?

Either way is fine though - just my first reaction to this function is that it's quite hard to follow - 70 lines and a lot of conditionals

Reply:
@ryanmccombe thanks for your thoughts. I've moved the exception handling to the param building function and tried to simplify the event tracking context. Let me know if this is better